Readme.org


* interface and tweaks
** exwm
   #+BEGIN_SRC emacs-lisp
;     (use-package exwm
;         :ensure t 
;         :config
;         (require 'exwm-config)
;         (exwm-config-default)
;         )
   #+END_SRC

** zerodark-theme
   #+BEGIN_SRC emacs-lisp
(use-package zerodark-theme
    :ensure t 
    :config
    (progn (zerodark-setup-modeline-format)
           ))
   #+END_SRC

** transparency
#+BEGIN_SRC emacs-lisp
;(set-frame-parameter (selected-frame) 'alpha '(85 . 50))
;(add-to-list 'default-frame-alist '(alpha . (85 . 50)))
 (defun toggle-transparency ()
   (interactive)
   (let ((alpha (frame-parameter nil 'alpha)))
     (set-frame-parameter
      nil 'alpha
      (if (eql (cond ((numberp alpha) alpha)
                     ((numberp (cdr alpha)) (cdr alpha))
                     ;; Also handle undocumented (<active> <inactive>) form.
                     ((numberp (cadr alpha)) (cadr alpha)))
               100)
          '(85 . 50) '(100 . 100)))))

#+END_SRC

** tweaks
#+BEGIN_SRC emacs-lisp 
  (tool-bar-mode -1)
  (setq inhibit-startup-message t)
  (global-set-key (kbd "<f5>") 'revert-buffer)
  (setq frame-title-format "%b")
  (desktop-save-mode)
  (delete-selection-mode 1)
                                          ;  (load-theme 'deeper-blue)
  (set-face-background 'region "azure4")
  (set-face-background 'cursor "sea green")
  (setq column-number-mode t) 
  (setq ring-bell-function 'ignore)
  (global-prettify-symbols-mode t)
  (add-hook 'prog-mode-hook (lambda () (linum-mode t)))


  (server-start)
  (put 'upcase-region 'disabled nil)
  (put 'downcase-region 'disabled nil)
  (add-to-list 'initial-frame-alist '(fullscreen . maximized))
  (autoload 'zap-up-to-char "misc"
    "Kill up to, but not including ARGth occurrence of CHAR.
      \(fn arg char)"
    'interactive)
  (global-set-key "\M-z" 'zap-up-to-char)
  (scroll-bar-mode -1)
  (setq-default truncate-lines 0)
#+END_SRC

** smex
  #+BEGIN_SRC emacs-lisp
(use-package smex
    :ensure t 
    :config
    (progn (smex-initialize)
           ))
  #+END_SRC

** flx-ido
  #+BEGIN_SRC emacs-lisp
    (use-package flx
      :ensure t )

    (use-package flx-ido
        :ensure t 
        :config
        (progn (flx-ido-mode +1)
               
               ))
  #+END_SRC

** beacon
  #+BEGIN_SRC emacs-lisp
    (use-package beacon
        :ensure t 
        :config
        (progn (beacon-mode 1)
               (setq beacon-push-mark 35)
;               (setq beacon-color "#666600")
               ))
  #+END_SRC

** atomic-chrome
   [[https://chrome.google.com/webstore/detail/atomic-chrome/lhaoghhllmiaaagaffababmkdllgfcmc/related?hl%3Den][atomic-chrome installation]]
  #+BEGIN_SRC emacs-lisp
(use-package atomic-chrome
    :ensure t 
    :config
    (progn (atomic-chrome-start-server)
           ))
  #+END_SRC

** hl-line-mode
#+BEGIN_SRC emacs-lisp
  (global-hl-line-mode)
  (set-face-background 'hl-line "#282a36")
#+END_SRC

** tabbar
#+BEGIN_SRC emacs-lisp
(use-package tabbar
  :ensure t
  :config
  (tabbar-mode 1))
#+END_SRC

** ace-window
#+BEGIN_SRC emacs-lisp
  (use-package ace-window
    :ensure t
    :init
    (progn
      (with-eval-after-load 'org
    (define-key org-mode-map [(control tab)] 'ace-window))
      (global-set-key (kbd "<C-tab>") 'ace-window)
      (setq aw-keys '(?j ?k ?l ))
      (custom-set-faces
       '(aw-leading-char-face
         ((t (:inherit ace-jump-face-foreground :height 2.0 :background "sea green" :foreground "#000" ))))
       ))) 

#+END_SRC

** undo-tree
#+BEGIN_SRC emacs-lisp
 (use-package undo-tree
    :ensure t
    :config (global-undo-tree-mode))
#+END_SRC

** flx
   #+BEGIN_SRC emacs-lisp
     (use-package flx
         :ensure t 
         )
   #+END_SRC

** nyan-mode
#+BEGIN_SRC emacs-lisp
  (use-package nyan-mode
    :ensure t
    :config
    (progn
      (nyan-mode t)
      (nyan-start-animation)
      (nyan-toggle-wavy-trail)
      (setq nyan-bar-length 10)
      (setq nyan-minimum-window-width 5)
      ))
#+END_SRC

** neotree
  #+BEGIN_SRC emacs-lisp
    (use-package neotree
        :ensure t 
        :config
        (progn (global-set-key [f8] 'neotree-toggle)
               (setq neo-theme (if (display-graphic-p) 'icons 'arrow))
               (setq split-window-preferred-function 'neotree-split-window-sensibly)
               ))
  #+END_SRC

** Shell
#+BEGIN_SRC emacs-lisp
  (global-set-key
   (kbd "C-x t")
   (lambda()(interactive)   
     (if ( or (get-buffer "*ansi-term*") (get-buffer "*eshell*") )
         (rename-uniquely))
     (if (string=  window-system "w32")
         (eshell)
       (ansi-term "/bin/bash")
       )))
#+END_SRC

** smartparens
#+BEGIN_SRC emacs-lisp
  (use-package smartparens
    :ensure t
    :config
    (progn
      (smartparens-global-mode 1)
      (show-smartparens-global-mode 1)))
#+END_SRC

** volatile-highlights
#+BEGIN_SRC emacs-lisp
  (use-package volatile-highlights
    :ensure t
    :config
    (volatile-highlights-mode t))
#+END_SRC

** My commands
#+BEGIN_SRC emacs-lisp
  (define-prefix-command 'accacio-map)
  (global-set-key (kbd "<escape><escape>") 'accacio-map)
  (define-key accacio-map (kbd "c")
   (defun myinit-customize ()
    (interactive)
    (setq ReadmeBuffer (or
                        (get-buffer "Readme.org")
                        (get-buffer "Readme.org<.emacs.d>")
                        ))
    (setq VisitedFile  (buffer-file-name ReadmeBuffer))
    (setq myReadme "~/.emacs.d/Readme.org")
    
    (if (string= VisitedFile (expand-file-name myReadme))
        (if (eq (get-buffer (current-buffer)) ReadmeBuffer)
            (kill-buffer ReadmeBuffer)
          (switch-to-buffer ReadmeBuffer)
          )
      (find-file "~/.emacs.d/Readme.org")
      )
    )
   )
  (define-key accacio-map (kbd "t") 'toggle-transparency)
  (define-key accacio-map (kbd "g") 'guru-global-mode)
  (define-key accacio-map (kbd "x") 'counsel-M-x)

#+END_SRC

** guru-mode
   #+BEGIN_SRC emacs-lisp
     (use-package guru-mode
         :ensure t 
         :config
         (progn ;(guru-global-mode)
                ))
   #+END_SRC

** all-the-icons
   #+BEGIN_SRC emacs-lisp
     (use-package all-the-icons
       :ensure t 
       :config
         (progn
           (unless (package-installed-p 'all-the-icons)
             (all-the-icons-install-fonts))
           ))
   #+END_SRC

** rich-minority
   #+BEGIN_SRC emacs-lisp
     (use-package rich-minority
         :ensure t 
         :config
         (progn
           (rich-minority-mode 1)

     (setq rm-blacklist
           (format "^ \\(%s\\)$"
                   (mapconcat #'identity
                              '( "Irony" "company" "HelmGtags" "ARev" "Hi"  "ElDoc"  "Abbrev" "SP"  "Fly.*"  "yas.*" "Projectile.*" "Undo-Tree" "PgLn" "WK" "ivy" "VHL" ".*\(\*\).*" )
                              "\\|")))
                ))
   #+END_SRC

** sublimity
#+BEGIN_SRC emacs-lisp
  (use-package sublimity
      :ensure t 
      :config
      (progn
        (require 'sublimity-map)
        (require 'sublimity-scroll)
        (sublimity-map-set-delay 1)
        (sublimity-mode)
        ))
#+END_SRC
* Navigation
** Ibuffer
#+BEGIN_SRC emacs-lisp
  (global-set-key (kbd "C-x C-b") 'ibuffer)
  (setq ibuffer-saved-filter-groups
        (quote (("default"
                 ("dired" (mode . dired-mode))
                 ("org" (mode . org-mode))
                 ("programming" (or (or (mode . c-mode) (mode . c++-mode) ) (mode . python-mode) ) )
                 ("emacs" (or (name . "^\\*scratch\\*$") (name . "^\\*Messages\\*$") (name . "^\\*Warnings\\*$")))
                 ("pdf" (name . "^.*pdf$"))
                 ("resto" (name . "^\\*.*\\*$"))
                 ))))
  (add-hook 'ibuffer-mode-hook
            (lambda ()
              (ibuffer-auto-mode 1)
              (ibuffer-switch-to-saved-filter-groups "default")))
#+END_SRC
** Dired
#+BEGIN_SRC emacs-lisp
  (put 'dired-find-alternate-file 'disabled nil)
  (eval-after-load "dired" '(progn (define-key dired-mode-map (kbd "RET") 'dired-find-alternate-file)))
  (setq dired-dwim-target t)

#+END_SRC

** Try 
#+BEGIN_SRC emacs-lisp
(use-package try
  :ensure t)
#+END_SRC

** which-key
#+BEGIN_SRC emacs-lisp
(use-package which-key
  :ensure t
  :config (which-key-mode))
#+END_SRC

** counsel
#+BEGIN_SRC emacs-lisp
  (use-package counsel
    :ensure t
    :bind
    ("M-y" . counsel-yank-pop))

#+END_SRC

** swiper
#+BEGIN_SRC emacs-lisp
  (use-package swiper
    :ensure t
    :config
    (progn
      (ivy-mode 1)
      (setq ivy-use-virtual-buffers t)
      (setq enable-recursive-minibuffers nil)
      (global-set-key "\C-s" 'swiper)
      (global-set-key (kbd "C-c C-r") 'ivy-resume)
      (global-set-key (kbd "<f6>") 'ivy-resume)
      (global-set-key (kbd "M-x") 'counsel-M-x)
      (global-set-key (kbd "C-x C-f") 'counsel-find-file)
      (global-set-key (kbd "<f1> f") 'counsel-describe-function)
      (global-set-key (kbd "<f1> v") 'counsel-describe-variable)
      (global-set-key (kbd "<f1> l") 'counsel-find-library)
      (global-set-key (kbd "<f2> i") 'counsel-info-lookup-symbol)
      (global-set-key (kbd "<f2> u") 'counsel-unicode-char)
      (global-set-key (kbd "C-c g") 'counsel-git)
      (global-set-key (kbd "C-c j") 'counsel-git-grep)
      (global-set-key (kbd "C-c k") 'counsel-ag)
      (global-set-key (kbd "C-x l") 'counsel-locate)
      (global-set-key (kbd "C-S-o") 'counsel-rhythmbox)
      (define-key minibuffer-local-map (kbd "C-r") 'counsel-minibuffer-history)
      ))
#+END_SRC

** avy
#+BEGIN_SRC emacs-lisp
  (use-package avy
    :ensure t
    :bind ("C-:" . avy-goto-char)
    :bind ("M-g g" . avy-goto-line )
    )
#+END_SRC

** iedit-mode
#+BEGIN_SRC emacs-lisp
  (use-package iedit
    :ensure t
    :config (define-key global-map (kbd "C-c ;") 'iedit-mode))
#+END_SRC
  
** Yasnippets
#+BEGIN_SRC emacs-lisp
 (use-package yasnippet
    :ensure t
    :config (yas-global-mode 1))
 (use-package yasnippet-snippets
    :ensure t)
#+END_SRC

** projectile
   #+BEGIN_SRC emacs-lisp
     (use-package projectile
         :ensure t 
         :config
         (progn (projectile-mode)
                ))

     (use-package counsel-projectile
       :ensure t
       :config
       (counsel-projectile-mode))
   #+END_SRC

   
** ag
   #+BEGIN_SRC emacs-lisp
     (use-package ag
         :ensure t 
         :config
         (progn 
           )
         )
   #+END_SRC
* Planning / Org-Mode
** org-bullets
#+BEGIN_SRC emacs-lisp
  (use-package org-bullets
    :ensure t
    :config
    (add-hook 'org-mode-hook (lambda () (org-bullets-mode 1))))

  (custom-set-variables '(org-hide-leading-stars t))
#+END_SRC
#+BEGIN_SRC emacs-lisp 
  (setq org-ellipsis "⤵") ;⤵ ⭍ ↲ ⮯
  (setq org-src-fontify-natively t)
  (setq org-src-tab-acts-natively t)
  (setq org-src-window-setup 'current-window)


#+END_SRC
   
* Reading
** pdf-tools
   #+BEGIN_SRC emacs-lisp
     (use-package pdf-tools
         :ensure t 
         :config
         (progn (pdf-tools-install)
                ))
   #+END_SRC
** engine-mode
   #+BEGIN_SRC emacs-lisp
     (use-package engine-mode
         :ensure t 
         :config
         (progn
           (defengine duckduckgo
             "https://duckduckgo.com/?q=%s"
             :keybinding "d")
           (defengine wikipedia
       "http://www.wikipedia.org/search-redirect.php?language=en&go=Go&search=%s"
       :keybinding "w")
           (defengine stack-overflow
             "https://stackoverflow.com/search?q=%s"
             :keybinding "s")
           (defengine github
             "https://github.com/search?ref=simplesearch&q=%s"
             :keybinding "g")
          
           (defengine youtube
       "https://www.youtube.com/results?search_query=%s")
           (engine-mode t) 
           )
         )
   #+END_SRC   
* Programming
** Org-babel c++
#+BEGIN_SRC emacs-lisp
  (org-babel-do-load-languages
   'org-babel-load-languages '((C . t)))
#+END_SRC

** magit
#+BEGIN_SRC emacs-lisp
  (use-package magit
    :ensure t
    :config
    (progn
      (set-default 'magit-stage-all-confirm nil)
      (add-hook 'magit-mode-hook 'magit-load-config-extensions)
      (global-unset-key (kbd "C-x g"))
      (global-set-key (kbd "C-x g h") 'magit-log)
      (global-set-key (kbd "C-x g f") 'magit-file-log)
      (global-set-key (kbd "C-x g b") 'magit-blame-mode)
      (global-set-key (kbd "C-x g m") 'magit-branch-manager)
      (global-set-key (kbd "C-x g c") 'magit-branch)
      (global-set-key (kbd "C-x g s") 'magit-status)
      (global-set-key (kbd "C-x g r") 'magit-reflog)
      (global-set-key (kbd "C-x g t") 'magit-tag)
      (setq user-full-name "Rafael Accácio Nogueira")
      (setq user-mail-address "raccacio@poli.ufrj.br")
      ))

#+END_SRC

** google-c-style flymake
#+BEGIN_SRC emacs-lisp
  (use-package flymake-cursor
    :ensure t)

  (defun my:flymake-google-init ()
    (require 'flymake-google-cpplint)
    (custom-set-variables
     '(flymake-google-cpplint-command
       (if (string=  window-system "w32")
           '"d:/Users/Accacio/AppData/Local/Programs/Python/Python36-32/Scripts/cpplint.exe"
         '"/usr/local/bin/cpplint"
         )
       )
     )
    (flymake-google-cpplint-load)
    (flymake-cursor-mode 1)
    )

  (use-package flymake-google-cpplint
    :ensure t
    :config
    (progn
      (add-hook 'c-mode-hook 'my:flymake-google-init)
      (add-hook 'c++-mode-hook 'my:flymake-google-init)
      ))

  (use-package google-c-style
    :ensure t
    :config 
    (progn
      (add-hook 'c-mode-common-hook 'google-set-c-style)
      (add-hook 'c-mode-common-hook 'google-make-newline-indent)
      ))

#+END_SRC

** flycheck
#+BEGIN_SRC emacs-lisp
  (use-package flycheck
    :ensure t
    :config
    (progn
                                          ;    (add-hook 'after-init-hook #'global-flycheck-mode)
      (add-hook 'prog-mode-hook 'flycheck-mode)
      ))
#+END_SRC

** company
#+BEGIN_SRC emacs-lisp
  (use-package company
    :ensure t
    :config
    (progn 
      (add-hook 'after-init-hook 'global-company-mode)
      (setq company-idle-delay 0)
      )
    )

  (use-package company-quickhelp
  :ensure t
  :config
  (progn
    (setq company-minimum-prefix-length 1)
                                          ;    (setq company-tooltip-maximum-width 20)
    (company-quickhelp-mode)
    ))

#+END_SRC
 workaround fci incompability

#+BEGIN_SRC emacs-lisp


(defvar-local company-fci-mode-on-p nil)

(defun company-turn-off-fci (&rest ignore)
  (when (boundp 'fci-mode)
    (setq company-fci-mode-on-p fci-mode)
    (when fci-mode (fci-mode -1))))

(defun company-maybe-turn-on-fci (&rest ignore)
  (when company-fci-mode-on-p (fci-mode 1)))

(add-hook 'company-completion-started-hook 'company-turn-off-fci)
(add-hook 'company-completion-finished-hook 'company-maybe-turn-on-fci)
(add-hook 'company-completion-cancelled-hook 'company-maybe-turn-on-fci)

#+END_SRC

** irony-mode
#+BEGIN_SRC emacs-lisp
      (use-package irony
        :ensure t
        :config 
        (progn
          (add-hook 'c++-mode-hook 'irony-mode)
          (add-hook 'c-mode-hook 'irony-mode)
          (add-hook 'objc-mode-hook 'irony-mode)
          
          (add-hook 'c++-mode-hook 'company-mode)
          (add-hook 'c-mode-hook 'company-mode)
          (add-hook 'objc-mode-hook 'company-mode)
          
          (add-hook 'after-init-hook 'global-company-mode)
          
          
          (add-hook 'irony-mode-hook 'irony-cdb-autosetup-compile-options)
          
          
          
          ))

      (use-package company-irony
        :ensure t
        :config
      (eval-after-load 'company
            '(add-to-list 'company-backends '(company-irony))))

      (use-package flycheck-irony
        :ensure t
        :config
        (eval-after-load 'flycheck
          '(add-hook 'flycheck-mode-hook #'flycheck-irony-setup)))


      ;; Windows performance tweaks
      ;;
      (when (boundp 'w32-pipe-read-delay)
        (setq w32-pipe-read-delay 0))
      ;; Set the buffer (setq )ize to 64K on Windows (from the original 4K)
      (when (boundp 'w32-pipe-buffer-size)
        (setq irony-server-w32-pipe-buffer-size (* 64 1024)))


  (use-package company-irony-c-headers
    :ensure t
    :config (eval-after-load 'company
              '(add-to-list
                'company-backends 
                '(company-irony-c-headers
                  company-irony
                                          ;company-yasnippet
                  ))))

  (use-package irony-eldoc
    :ensure t
    :config
    (add-hook 'irony-mode-hook 'irony-eldoc))
#+END_SRC

** idle-highlights
#+BEGIN_SRC emacs-lisp
  (use-package idle-highlight-mode
    :ensure t
    :config
    (progn  (add-hook 'prog-mode-hook (lambda () (idle-highlight-mode t)))
            (set-face-background 'idle-highlight "#484a56")
            ))
#+END_SRC

** fill-column-indicator
#+BEGIN_SRC emacs-lisp
  (use-package fill-column-indicator
    :ensure t 
    :config
    (progn (setq-default fill-column 80)
           (add-hook 'prog-mode-hook (lambda () (fci-mode t)))
           ))


#+END_SRC

** cmake-mode
  #+BEGIN_SRC emacs-lisp
(use-package cmake-mode
    :ensure t 
    :config
    (progn (setq auto-mode-alist
      (append
       '(("CMakeLists\\.txt\\'" . cmake-mode))
       '(("\\.cmake\\'" . cmake-mode))
       auto-mode-alist))

(autoload 'cmake-mode "~/.emacs.d/elpa/cmake-mode-20180104.1137/cmake-mode-autoloads.el" t)
           ))
  #+END_SRC

** helm-gtags
  #+BEGIN_SRC emacs-lisp
    (use-package helm-gtags
      :ensure t 
      :config
      (progn (add-hook 'c++-mode-hook 'helm-gtags-mode)
             (add-hook 'c-mode-hook 'helm-gtags-mode)
             (global-set-key (kbd "C-.") 'helm-gtags-dwim)
             ))
  #+END_SRC

** multiple-cursors
  #+BEGIN_SRC emacs-lisp
    (use-package multiple-cursors
        :ensure t 
        :init
        (progn (global-set-key (kbd "C-x <down>") 'mc/mark-next-like-this)
               (global-set-key (kbd "C-x <up>") 'mc/mark-previous-like-this)
               ))
  #+END_SRC

** ace-mc
   #+BEGIN_SRC emacs-lisp
     (use-package ace-mc
         :ensure t 
         :config
         (progn (global-set-key (kbd "C-0") 'ace-mc-add-multiple-cursors)
                ))
   #+END_SRC
** expand-region
  #+BEGIN_SRC emacs-lisp
    (use-package expand-region
        :ensure t 
        :init
        (progn (global-set-key (kbd "C-=") 'er/expand-region)
               ))
  #+END_SRC
* HTML
** rainbow-mode
  #+BEGIN_SRC emacs-lisp
    (use-package rainbow-mode
        :ensure t 
        :config
        (progn (add-hook 'html-mode-hook 'rainbow-mode)
               (add-hook 'css-mode-hook 'rainbow-mode)
               ))
  #+END_SRC

* Games
** 2048
   #+BEGIN_SRC emacs-lisp
     (use-package 2048-game
       :ensure t 
         )
   #+END_SRC

* Finances 
** ledger-mode
   #+BEGIN_SRC emacs-lisp
     (use-package ledger-mode
         :ensure t 
         :init
         (setq ledger-clear-whole-transactions 1)
         :config
         (progn (add-to-list 'auto-mode-alist '("\\.ledger$" . ledger-mode))
                )
         )
   #+END_SRC


* Latex
** auctex
#+BEGIN_SRC emacs-lisp
;; (require 'auctex)
     (use-package auctex
     :defer t
     :ensure t)
(setq TeX-auto-save t)
(setq TeX-parse-self t)
(setq TeX-save-query nil)
(defun turn-on-outline-minor-mode ()
(outline-minor-mode 1))

(add-hook 'LaTeX-mode-hook 'turn-on-outline-minor-mode)
(add-hook 'latex-mode-hook 'turn-on-outline-minor-mode)
(setq outline-minor-mode-prefix "\C-c \C-o") ; Or something else
;(setq TeX-PDF-mode t)

(use-package company-auctex
    :ensure t 
    :config
    (progn 
           ))
(company-auctex-init)

#+END_SRC 
** company-bibtex
#+BEGIN_SRC emacs-lisp
(use-package company-bibtex
    :ensure t 
    :config
    (progn 
           ))

#+END_SRC

#+BEGIN_SRC emacs-lisp
(use-package company-reftex
    :ensure t 
    :config
    (progn 
           ))

#+END_SRC

* Blog
#+BEGIN_SRC emacs-lisp
(setq org-publish-project-alist
      '(

  ("org-accacio"
          ;; Path to your org files.
          :base-directory "~/site/org/"
          :base-extension "org"

          ;; Path to your Jekyll project.
          :publishing-directory "~/site/meublog/"
          :recursive t
          :publishing-function org-html-publish-to-html
          :headline-levels 4
          :html-extension "html"
          :body-only t ;; Only export section between <body> </body>
    )


    ("org-static-acc"
          :base-directory "~/site/org/"
          :base-extension "css\\|js\\|png\\|jpg\\|gif\\|pdf\\|mp3\\|ogg\\|swf\\|php"
          :publishing-directory "~/site/"
          :recursive t
          :publishing-function org-publish-attachment)

    ("acc" :components ("org-accacio" "org-static-acc"))

))

#+END_SRC


eshell/history
ls
ll
ls -a
ls
ll
doxygen 
PS1='\[\033[01;32m\]\u:\[\033[01;34m\]\w\[\033[00m\]\$' 
PS1
$PS1
whoami 
\	w
\w 
\u 
PS1='${debian_chroot:+($debian_chroot)}\u@\h:\w\$ ' 
PS1='${debian_chroot:+($debian_chroot)}\u@\h:\w\$ '
PS1='${debian_chroot:+($debian_chroot)}\u@\h:\w\$ '
echo $PS1
echo $PS1 |less -E
~/.bashrc
source ~/.bashrc
echo $PS1
/bin/bash
doxygen 
/bin/ls
ls
ll
ls -h
help ls
man ls
ls -a
ls
ll
ls
ll
alias ll=ls -l
ll
ls -l
alias
ll= ls -l
ll
ll
ll
ll
ll
clear

clc
clc
clc
clc

clc

ll
PS1='\[\033[01;32m\]\u:\[\033[01;34m\]\w\[\033[00m\]\$'
PS1
$PS1
export PS1='\[\033[01;32m\]\u:\[\033[01;34m\]\w\[\033[00m\]\$'
$PS1
. ~/.bashrc
. ~/.bashrc
. ~/.bashrc
ll
clc

ls
cd Documents/
ls
mkdir teste
cd teste/
ls
touch {a-z}.png
ls
touch [a-z].png
ls
ls
mkdir
mkdir build
cd build/
cmake ..
make
make ..
ls
cd ..
ls
ls
ls build/
clc

clc

clear

clc

cmake
cmake .
make
cmake .
make
./matgc 
cmake . & make
cmake .
make


init.el

(require 'package)
(setq package-enable-at-startup nil)
(add-to-list 'package-archives
	     '("melpa" . "https://melpa.org/packages/")
	     '("elpa" . "https://elpa.org/packages/"))
(add-to-list 'package-archives
	     '("melpa-stable" . "https://stable.melpa.org/packages/"))


(package-initialize)

(unless (package-installed-p 'use-package)
  (package-refresh-contents)
  (package-install 'use-package))

(org-babel-load-file (expand-file-name "~/.emacs.d/Readme.org"))

(custom-set-variables
 ;; custom-set-variables was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 '(flymake-google-cpplint-command
   (if
       (string= window-system "w32")
       (quote "d:/Users/Accacio/AppData/Local/Programs/Python/Python36-32/Scripts/cpplint.exe")
     (quote "/usr/local/bin/cpplint")))
 '(ledger-reports
   (quote
    (("reg" "ledger [[ledger-mode-flags]] -f /home/accacio/.emacs.d/ledger.ledger reg")
     (#("bal" 0 1
	(idx 0))
      "%(binary) -f %(ledger-file) bal")
     (#("payee" 0 1
	(idx 2))
      "%(binary) -f %(ledger-file) reg @%(payee)")
     (#("account" 0 1
	(idx 3))
      "%(binary) -f %(ledger-file) reg %(account)"))))
 '(minimap-dedicated-window t)
 '(minimap-hide-scroll-bar nil)
 '(minimap-mode nil)
 '(minimap-recreate-window nil)
 '(minimap-window-location (quote left))
 '(neo-autorefresh t)
 '(neo-click-changes-root t)
 '(neo-confirm-change-root (quote off-p))
 '(neo-force-change-root t)
 '(neo-smart-open t)
 '(org-hide-leading-stars nil)
 '(package-selected-packages
   (quote
    (ag ledger-mode ace-mc sublimity counsel-projectile projectile pdf-tools helm-company atomic-chrome beacon flx-ido smex helm-gtags rainbow-mode irony-eldoc expand-region multiple-cursors neotree cmake-mode fill-column-indicator idle-highlight-mode idle-highligh-mode ox-reveal smartparens company-irony-c-headers company-irony irony irony-mode nyan-mode company flymake-cursor google-c-style flymake-google-cpplint flycheck undo-tree yasnippet yasnippet-snippet yasnippets iedit magit which-key use-package try tabbar org-bullets counsel ace-window))))
(custom-set-faces
 ;; custom-set-faces was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 '(aw-leading-char-face ((t (:inherit ace-jump-face-foreground :height 2.0 :background "sea green" :foreground "#000")))))
